locking.go

```diff
@@ -185,7 +205,7 @@ func NewLockInstance(ctxParent context.Context, accountName, accountKey, lockNam
 		lockInstance.used = true
```

Comment:
Is this thread safe? Should we use a mutex, channel or atomic uint32?

Reply:
Yeah this should be done - good point!

Reply:
same for lockAcquired?

Reply:
Added a mutex

Reply:
Do you think its right - I now lock on any of the operations so you can't 'Renew' while your already unlocking etc.

Reply:
Looks ok - do you need a test to confirm?

Reply:
I'll take a crack at a test.... gonna be a fun one to write!